assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -1232,5 +1243,311 @@ describe("DiversionPage", () => {
+          name: RPCRadioButtonName(selectedRoutePattern),
+        })
+      ).toBeChecked()
+    })
+
+    test("when route is changed, selects first 'inbound' route pattern first", async () => {
+      const routes = routeFactory.buildList(2)
+      const [route1, route2] = routes
+
+      const inboundRoutePatterns = routePatternFactory.buildList(2, {
+        routeId: route2.id,
+        directionId: 1,
+      })
+      const outboundRoutePatterns = routePatternFactory.buildList(2, {
+        routeId: route2.id,
+        directionId: 0,
+      })
+      const routePatterns = outboundRoutePatterns.concat(inboundRoutePatterns)
+
+      const [selectedRoutePattern] = inboundRoutePatterns
+
+      jest.mocked(fetchRoutePatterns).mockResolvedValue(routePatterns)
+
+      render(
+        <RoutesProvider routes={routes}>
+          <DiversionPage
+            originalRoute={{
+              route: route1,
+              routePattern: routePatternFactory.build({ routeId: route1.id }),
+            }}
+          />
+        </RoutesProvider>
+      )
+
+      await userEvent.click(
+        screen.getByRole("button", { name: "Change route or direction" })
+      )
+
+      await userEvent.selectOptions(
+        screen.getByRole("combobox", { name: "Choose route" }),
+        route2.name
+      )
+
+      expect(screen.getByRole("radio", { name: "Inbound" })).toBeChecked()
+
+      await waitFor(() => {
+        expect(
+          screen.getByRole("radio", {
+            name: RPCRadioButtonName(selectedRoutePattern),
+          })
+        ).toBeChecked()
+      })
+    })
+
+    test("when route is changed, selects first route pattern otherwise", async () => {
+      const routes = routeFactory.buildList(2)
+      const [route1, route2] = routes
+
+      const outboundRoutePatterns = routePatternFactory.buildList(2, {
+        routeId: route2.id,
+        directionId: 0,
+      })
+      const routePatterns = outboundRoutePatterns
+
+      const [selectedRoutePattern] = outboundRoutePatterns
+
+      jest.mocked(fetchRoutePatterns).mockResolvedValue(routePatterns)
+
+      render(
+        <RoutesProvider routes={routes}>
+          <DiversionPage
+            originalRoute={{
+              route: route1,
+              routePattern: routePatternFactory.build({ routeId: route1.id }),
+            }}
+          />
+        </RoutesProvider>
+      )
+
+      await userEvent.click(
+        screen.getByRole("button", { name: "Change route or direction" })
+      )
+
+      await userEvent.selectOptions(
+        screen.getByRole("combobox", { name: "Choose route" }),
+        route2.name
+      )
+
+      expect(screen.getByRole("radio", { name: "Outbound" })).toBeChecked()
+
+      await waitFor(() => {
+        expect(
+          screen.getByRole("radio", {
+            name: RPCRadioButtonName(selectedRoutePattern),
+          })
+        ).toBeChecked()
+      })
+    })
   })
 })
+
+function RPCRadioButtonName(rp2: RoutePattern) {
```

Comment:
Nit: Should this be `const rpcRadioButtonName = (rp2: RoutePattern) => {...}` instead (and probably live at the top of the file 😩)?

Other nit: The variable name should perhaps just be `rp`?

---

Also...

I also would sooner see the tests be a bit more explicit about their data, so something like

```typescript
rp = routePatternFactory.build({
  name: "Harvard via Allston",
  description: "from Nubian Station",
})

/* ... */

screen.getByRole("radio", {
  name: "Harvard via Allston from Nubian Station",
})
```

That way the test is super explicit about what it's testing (and it's clearer what type of thing we expect `name` and `description` to be). As it currently stands, in order to figure out what the assertion actually is, we have to trace through `RPCRadioButtonName`, and if `patternDisplayName` is broken then the assertion will assert the broken/wrong thing.

This can be a follow-up, and doesn't need to block merging this PR.

Reply:
Does it make it any better that `patternDisplayName` is tested? https://github.com/mbta/skate/blob/main/assets/tests/components/mapPage/routePropertiesCard.test.tsx#L58-L104

Reply:
Part of it https://github.com/mbta/skate/compare/356f1eb5c81370545b67bc2e28304b7595914805..aeb951f1a4020482c2a6040753fe5bfc440d5de2